### PR TITLE
feature/6988-roettger-AddDeprecationNoticePackage

### PR DESCRIPTION
**Pull request by TimRoe**
## Description of Change

Incorporating the design system linting package that will issue deprecation notices within the app for using outdated components--will be used for nudging retirement of flagship components as well as if any components within the mobile components package are deprecated. In this initial version, it includes the VAButton->Button transition.

## Testing

- [x] Tested on iOS  (just smoke tested it still ran)

- Ran `yarn start` to confirming it will annoy devs into heeding the deprecation notices by spamming them whenever you're running react-native as well as appropriately flagging them with warnings in the files
- Ran `yarn lint` to confirm it also includes the deprecation notices. 
- Ran `yarn lint:ci` to confirm it **_does not_** include deprecation notices as it would cause the build jobs to fail over them

Nothing to QA for this ticket, merely repo configuration change.

## PR Checklist

  **Reviewer:** Confirm the items below as you review

- [ ] PR is connected to issue(s)
- [ ] Tests are included to cover this change (when possible)
- [ ] No magic strings (All string unions follow the [Union -> Constant](https://github.com/department-of-veterans-affairs/va-mobile-app/blob/develop/VAMobile/src/constants/common.ts) type pattern)
- [ ] No secrets or API keys are checked in
- [ ] All imports are absolute (no relative imports)
- [ ] New functions and Redux work have proper TSDoc annotations

### CODE REVIEW

**Review by Sparowhawk — APPROVED** [review]

**Review by narin — DISMISSED** [review]

**Review by narin — APPROVED** [review]

**Comment by narin** [comment]
@Sparowhawk can you please re-approve? Resolved a conflict in `package.json`

**Review by theodur — DISMISSED** [review]

**Comment by theodur** [comment]
@narin @TimRoe Should be good to merge now

**Review by narin — APPROVED** [review]

**Comment by narin** [comment]
This PR doesn't need a QA review as mentioned in the description. It can be merged once it gets 2 approvals.

**Review by theodur — APPROVED** [review]

**Pull request merged**
